### PR TITLE
Add proper statusCode for duplicate

### PR DESCRIPTION
**Pull request by loay**
### Description
Add proper statusCode for http client error of Duplicate

#### Related issues

connect to https://github.com/strongloop/loopback/issues/2512

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style

### CODE REVIEW

**Comment by loay** [comment]
@bajtos @jannyHou PTAL. Thanks.

**Comment by loay** [comment]
No it does not use status either ..  for example, in mysql example:
`'ER_DUP_ENTRY', errno: 1062, sqlState: '23000', index: 0 `
So I am not sure if I should be using the test case modification after all !

**Comment by bajtos** [comment]
+1 for returning 409 instead of 500 when a duplicate id is detected in CREATE operation.

I think we need to fix all connectors to detect the duplicate entry and report a correct error back before we can land the change in the shared test suite.

File: lib/connectors/memory.js [hunk 1]
```diff
@@ -220,8 +220,12 @@ Memory.prototype._createSync = function(model, data, fn) {
     this.collection(model, {});
   }
 
-  if (this.collection(model)[id])
-    return fn(new Error(g.f('Duplicate entry for %s.%s', model, idName)));
+  if (this.collection(model)[id]) {
+    var errMsg = g.f('Duplicate entry for %s.%s', model, idName);
+    var error = new Error(errMsg);
```
Comment (bajtos) [inline lib/connectors/memory.js]: Nitpick - feel free to ignore: I would personally simplify this lines to a single statement.

```js
var error = new Error(g.f(
  'Duplicate entry for %s.%s', 
  model, idName));
```

**Comment by bajtos** [comment]
Alternatively, you can move the test for status code to memory-connector tests only, so that we can land this fix sooner and defer fixing other connectors for later.

**Pull request merged**